service/src/main/java/org/cbioportal/genome_nexus/service/internal/CancerHotspotServiceImpl.java

```diff
@@ -156,25 +159,28 @@ public List<Hotspot> getHotspotAnnotationsByGenomicLocation(String genomicLocati
         throws VariantAnnotationNotFoundException, VariantAnnotationWebServiceException,
         CancerHotspotsWebServiceException
     {
-        GenomicLocation location = this.notationConverter.parseGenomicLocation(genomicLocation);
-
-        return this.getHotspotAnnotationsByVariant(this.notationConverter.genomicToHgvs(location));
+        VariantAnnotation variantAnnotation = genomicLocationAnnotationService.getAnnotation(genomicLocation);
+        return this.getHotspotAnnotations(variantAnnotation);
     }
 
     @Override
     public List<AggregatedHotspots> getHotspotAnnotationsByGenomicLocations(List<GenomicLocation> genomicLocations)
         throws CancerHotspotsWebServiceException
     {
-        // convert genomic location to hgvs notation (there is always 1-1 mapping)
-        Map<String, GenomicLocation> variantToGenomicLocation = notationConverter.genomicToHgvsMap(genomicLocations);
+        List<VariantAnnotation> variantAnnotations = this.genomicLocationAnnotationService.getAnnotations(genomicLocations);
 
-        // query hotspots service by variant
-        List<AggregatedHotspots> hotspots = this.getHotspotAnnotationsByVariants(
-            new ArrayList<>(variantToGenomicLocation.keySet()));
-
-        // add genomic location info too
-        for (AggregatedHotspots aggregatedHotspots: hotspots) {
-            aggregatedHotspots.setGenomicLocation(variantToGenomicLocation.get(aggregatedHotspots.getVariant()));
+        Map<String, GenomicLocation> variantToGenomicLocation = new HashMap<>();
+        for (GenomicLocation genomicLocation: genomicLocations) {
+            variantToGenomicLocation.put(genomicLocationAnnotationService.getVariantFormat(genomicLocation), genomicLocation);
+        }
```

Comment:
We can use stream to simplify this, something like
```suggestion
Map<String, GenomicLocation> variantToGenomicLocation = genomicLocations.stream()
    .collect(Collectors.toMap(
        genomicLocation -> genomicLocationAnnotationService.getVariantFormat(genomicLocation),
        v -> v)
    );
```